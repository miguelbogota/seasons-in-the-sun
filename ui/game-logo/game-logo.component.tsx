import clsx from 'clsx';

export type GameLogoProps = {
  className?: string;
  style?: React.CSSProperties;
};

/**
 * GameLogo component.
 */
export function GameLogo(props: GameLogoProps) {
  const { className, style } = props;

  const rootClassName = clsx('game-logo', className);

  return (
    <h1 className={rootClassName} style={style}>
      <span>Seasons in</span>
      <span>the Sun</span>
    </h1>
  );
}
