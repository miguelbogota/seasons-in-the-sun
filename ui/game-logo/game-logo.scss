@mixin apply {
  h1.game-logo {
    font-size: var(--size-1);
    font-weight: var(--weight-bold);
    margin-bottom: var(--space-3);
    text-transform: uppercase;
    word-spacing: 6px;

    & > span {
      display: block;

      &:first-child {
        font-size: 0.72em;
        margin-bottom: -0.3em;
      }
    }
  }
}
