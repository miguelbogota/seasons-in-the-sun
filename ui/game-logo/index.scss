@use './game-logo' as game-logo;

@mixin apply {
  @include game-logo.apply;
}
