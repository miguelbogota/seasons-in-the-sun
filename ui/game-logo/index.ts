export { GameLogo, type GameLogoProps } from './game-logo.component';
