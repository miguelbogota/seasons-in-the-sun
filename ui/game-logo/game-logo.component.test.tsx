import { render } from '@testing-library/react';

import { GameLogo } from './game-logo.component';

describe('UI / GameLogo', () => {
  it('should match snapshot', () => {
    const { container } = render(<GameLogo />);
    expect(container).toMatchSnapshot();
  });
});
