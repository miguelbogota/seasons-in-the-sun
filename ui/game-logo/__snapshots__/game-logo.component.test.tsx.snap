// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UI / GameLogo > should match snapshot 1`] = `
<div>
  <h1
    class="game-logo"
  >
    <span>
      Seasons in
    </span>
    <span>
      the Sun
    </span>
  </h1>
</div>
`;
